packages/core-plugin-api/src/plugin-context/context.tsx

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright 2022 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+  createVersionedContext,
+  createVersionedValueMap,
+} from '@backstage/version-bridge';
+import React, { PropsWithChildren, useContext } from 'react';
+
+import { PluginContext } from './types';
+
+const PluginReactContext =
+  createVersionedContext<{ 1: PluginContext[] }>('plugin-context');
+
+/**
+ * A hook that gets the plugin context stack in the React component tree.
+ *
+ * @public
+ */
+export const usePluginContextStack = (): PluginContext[] | undefined => {
```

Comment:
The stack idea is interesting. Feels sensible but at the same time a bit scary as it might lead to some unnecessary coupling. How come you ended up exposing them that way?